### PR TITLE
Fix/kubernetes fixes and performance

### PR DESCRIPTION
**Pull request by dymbe**
- add caching
- add /ready-endpoint
- fix wrong names in k8s-template

### CODE REVIEW

File: geocoder_api/routers/geocoding.py [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 
 @router.get(
     "/",
     description="Returns a GeoJSON FeatureCollection of places matching the search query",
 )
-async def geocode(q: Annotated[str, Query(description="Search query")]):
+@cache(expire=3600)  # Expires after 1 hour
+async def geocode(
+    q: Annotated[str, Query(description="Search query")]
+) -> FeatureCollection:
```
Comment (kstigen) [inline geocoder_api/routers/geocoding.py:36]: This FeatureCollection does not provide any specification about what type of properties the returned geojson-features have?

Is it possible to also model them, so we can have a full documentation of the returned data?

**Review by kstigen — APPROVED** [review]

**Comment by dymbe** [comment]
Ended up creating my own models, as adding examples to the `geojson-pydantic`-models was not trivial, and they did not show up correctly in the OpenAPI-spec.

Also ended up exposing and documenting some additional query-parameters

File: geocoder_api/routers/geocoding.py [hunk 2]
```diff
@@ -1,38 +1,72 @@
...
-        params = {"lon": lon, "lat": lat}
+        # Remove None-parameters
+        params = {
+            k: v
+            for k, v in {"lon": lon, "lat": lat, **shared_params}.items()
+            if v is not None
+        }
         res = await client.get(f"{settings.photon_url}/reverse", params=params)
         if res.status_code == 200:
             return res.json()
```
Comment (kstigen) [inline geocoder_api/routers/geocoding.py:70]: We are not confident that the data we return here actually follows the model that you have specified in FeatureCollection?

Comment (dymbe) [reply geocoder_api/routers/geocoding.py:70]: If we get a 200, it should follow the model defined in FeatureCollection

**Review by kstigen — APPROVED** [review]

**Pull request merged**